### PR TITLE
[Remark 11] Remove unintentional `>` in text that comes from ancestor blockquote marker

### PR DESCRIPTION
**Pull request by seiyab**
## Description
Sadly, There's no proof that it won't cause other problems. Anyway, I explored edge cases and reviewed commonmark spec.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/language-markdown/print/preprocess.js [hunk 1]
```diff
@@ -214,6 +209,16 @@ function splitTextIntoSentences(ast) {
       };
     }
 
+    if (paragraphNode) {
+      const inBlockQuote = parentStack
+        .slice(paragraphIndex + 1)
+        .some((ancestor) => ancestor?.type === "blockquote");
+
+      if (inBlockQuote) {
+        text = text.replaceAll(/\n( *>[ \t]?)+/gu, "\n");
```
Comment (fisker) [inline src/language-markdown/print/preprocess.js]: Is it possible to get indentation from the second line, the rest line should have the same prefix?

Comment (fisker) [reply src/language-markdown/print/preprocess.js]: ignored node print should have the same problem.

Comment (fisker) [reply src/language-markdown/print/preprocess.js]: Maybe we can extract something like `getBlockquoteRaw` to reuse.

Comment (seiyab) [reply src/language-markdown/print/preprocess.js]: Indentation can vary: https://spec.commonmark.org/dingus/?text=%20%20%3E%20%20%20abc%0A%20%3E%20ab%0A%20%20%20%3E%20de
Or do I miss your intent?

Comment (fisker) [reply src/language-markdown/print/preprocess.js]: > Indentation can vary:

Didn't know...

**Comment by seiyab** [comment]
Reviewing by myself, I found I might be still somewhat confused. I'll reopen it refining the code and reflecting your review.

File: src/language-markdown/print/preprocess.js (hunk 1 above)
Comment (fisker) [reply src/language-markdown/print/preprocess.js]: Looks like line can't start with `>`? So we can find the `>` by the line text from value.

Comment (seiyab) [reply src/language-markdown/print/preprocess.js]: > Looks like line can't start with `>` ?

AFAIK, that's true. Line can't start with `>` since it starts blockquote interrupting paragraph.
On the other hand, now I found a new edge case, line can start with leading spaces + `>`.
```
This paragraph
    > is NOT interrupted by `>`.
```
It's not blockquote because 4+ spaces can't precede blockquote markser. It's not indented code block because it's not separated by any blank lines.

> So we can find the `>` by the line text from value.

Maybe I miss your point. But maybe we can find `>` in the edge case above by the line text from value.
